packages/perennial/test/verification/core/verifyCore.test.ts

```diff
@@ -0,0 +1,199 @@
+import { expect } from 'chai'
+import HRE from 'hardhat'
+import { constants, utils } from 'ethers'
+import { Deployment } from 'hardhat-deploy/types'
+import {
+  Collateral,
+  Collateral__factory,
+  Controller,
+  Controller__factory,
+  Forwarder,
+  Forwarder__factory,
+  Incentivizer,
+  Incentivizer__factory,
+  PerennialLens,
+  PerennialLens__factory,
+  ProxyAdmin,
+  ProxyAdmin__factory,
+  TimelockController,
+  TimelockController__factory,
+  UpgradeableBeacon,
+  UpgradeableBeacon__factory,
+} from '../../../types/generated'
+import { getMultisigAddress } from '../../../../common/testutil/constants'
+
+const { ethers } = HRE
+
+describe('Core - Mainnet Verification', () => {
+  let deployments: { [name: string]: Deployment }
+  let controller: Controller
+  let collateral: Collateral
+  let incentivizer: Incentivizer
+  let proxyAdmin: ProxyAdmin
+  let upgradeableBeacon: UpgradeableBeacon
+  let timelock: TimelockController
+  let forwarder: Forwarder
+  let lens: PerennialLens
+
+  beforeEach(async () => {
+    const [signer] = await ethers.getSigners()
+    deployments = await HRE.deployments.all()
+    controller = Controller__factory.connect(deployments['Controller_Proxy'].address, signer)
+    collateral = Collateral__factory.connect(deployments['Collateral_Proxy'].address, signer)
+    incentivizer = Incentivizer__factory.connect(deployments['Incentivizer_Proxy'].address, signer)
+    proxyAdmin = ProxyAdmin__factory.connect(deployments['ProxyAdmin'].address, signer)
+    upgradeableBeacon = UpgradeableBeacon__factory.connect(deployments['UpgradeableBeacon'].address, signer)
+    timelock = TimelockController__factory.connect(deployments['TimelockController'].address, signer)
+    forwarder = Forwarder__factory.connect(deployments['Forwarder'].address, signer)
```

Comment:
follow up: we can now safely remove forwarder from the codebase I think -- not in this PR though